src/Raven.Server/Documents/Patch/PatchDocumentCommand.cs

```diff
@@ -448,24 +449,39 @@ public class PatchDocumentCommand : PatchDocumentCommandBase
             bool isTest,
             bool debugMode,
             bool collectResultsNeeded,
-            bool returnDocument) : base(context, skipPatchIfChangeVectorMismatch, patch, patchIfMissing, createIfMissing, database, isTest, debugMode, collectResultsNeeded, returnDocument)
+            bool returnDocument,
+            bool ignoreMaxStepsForScript =false) : base(context, skipPatchIfChangeVectorMismatch, patch, patchIfMissing, createIfMissing, database, isTest, debugMode, collectResultsNeeded, returnDocument)
         {
             _id = id;
             _expectedChangeVector = expectedChangeVector;
-
+            _ignoreMaxStepsForScript = ignoreMaxStepsForScript;
             if (string.IsNullOrEmpty(id) || id.EndsWith(database.IdentityPartsSeparator) || id.EndsWith('|'))
                 throw new ArgumentException($"The ID argument has invalid value: '{id}'", nameof(id));
         }
         
         protected override long ExecuteCmd(DocumentsOperationContext context)
         {
             ScriptRunner.SingleRun runIfMissing = null;
+            int originalMaxStepsForScript = context.DocumentDatabase.Configuration.Patching.MaxStepsForScript;
 
             using (_database.Scripts.GetScriptRunner(_patch.Run, readOnly: false, out var run))
-            using (_patchIfMissing.Run != null ? _database.Scripts.GetScriptRunner(_patchIfMissing.Run, readOnly: false, out runIfMissing) : (IDisposable)null)
```

Comment:
check 'DisableMaxStatements' extension from JintExtensions

Reply:
Done

Reply:
Please check again, this returns IDisposable and on dispose it sets the old value again. So the proper usage should be:
```csharp
using ( run.ScriptEngine.DisableMaxStatements())
{
}
```

and then you do not need to do 'try/finally'

Reply:
Done